include/mariadb++/types.hpp

```diff
@@ -77,4 +78,17 @@ namespace mariadb {
     typedef std::shared_ptr<std::istream> stream_ref;
 }
 
+#ifdef MARIADB_BASE_VERSION
```

Comment:
Can you shorten the preprocessor ifs to a single line of `#if defined(MARIADB_BASE_VERSION) && MYSQL_VERSION_ID > foo`